Plugins/DotNET/DotNETExports.cpp

```diff
@@ -150,6 +146,42 @@ static void RegisterHandlers(AllHandlers* handlers, unsigned size)
         });
 }
 
+static void RegisterHandlers(AllHandlers* handlers, unsigned int size)
+{
+    if (size > sizeof(*handlers))
+    {
+        LOG_ERROR("RegisterHandlers argument contains too many entries, aborting");
+        return;
+    }
+    if (size < sizeof(*handlers))
+    {
+        LOG_WARNING("RegisterHandlers argument missing some entries - Managed/Unmanaged mismatch, update managed code");
+    }
+
+    LOG_INFO("Registering managed code handlers.");
+    s_handlers = *handlers;
+
+    if (s_handlers.MainLoop)
+    {
+        RegisterMainLoopHandler();
+    }
```

Comment:
I don't really feel too strongly about styles here, but:
* Please keep nonfunctional refactors to individual commits. Whether these are inlined or kept in separate functions is a pure preference, and we can discuss which is better, but it makes the functional part harder to review.
* If you are insisting on keeping it as a separate function, then the null checks should be in those functions, not here. And the functions should be passed as arguments, not globals.

Reply:
I'll break out the changes, and hopefully find a middle ground. I was called out on early PRs for making too many commits :P